### PR TITLE
protect cmd_free, cmd_exe and cmd_ready

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

## Migrations
NO

## Description
protect public API's in case of cmd_init() is not called before.

**WARNING:** This hides application side issue where it tries to use cmd library without initialization.

## Todos
- [x] Tests

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -174,7 +174,7 @@ def yottaBuildStep(target, compilerLabel) {
               setBuildStatus('PENDING', "build ${exampleName}", 'build starts')
               try {
                 // try to compile different configurations
-                execute("make CFLAGS=-DMBED_CONF_CMDLINE_USE_MINIMUM_SET=0")
+                execute("make CFLAGS=-DMBED_CONF_CMDLINE_USE_MINIMUM_SET=1")
```
Comment (tommikas) [inline Jenkinsfile]: This seems unrelated. Intentional?

Comment (jupe) [reply Jenkinsfile]: well, not directly, but undirectly since it test that minimum set can be compiled without compiler warnings....

Comment (jupe) [reply Jenkinsfile]: note that this was originally typo - it should be verify minimum set but it did not...
